### DISCUSSION TITLE
nfnl + fennel-language-server is it possible to get full completions for nvim lua builtins and libraries?

### DISCUSSION THREAD

**Discussion by andrewcrook** [Olical/nfnl#38] [Q&A]
First time using nfnl and fennel.
I have got automatic trans-compiling and completions working.
However, completions aren't quite there yet.

I wonder if there is a way of possibly getting completions  for vim.* and existing lua libraries to be a part of fennels completions via the fennel-language-server?

I thought I read it mentioned that aniseed can make lua visible through fennel  therefore I presumed could be use by the LSP to expand completions. eg the form of (vim.api.etc x y z)

is this not the case?
does anyone know of some solutions ?

**Comment by Olical** [comment]
I _think_ this is what does the trick for me https://github.com/Olical/dotfiles/blob/d21a2f20610a62ed6dae5b0fe9695dd4f2bcaf02/stowed/.config/nvim/fnl/plugins/mason.fnl#L15-L23
